sdk/core/azure-core/src/main/java/com/azure/core/implementation/RestProxy.java

```diff
@@ -154,6 +160,36 @@ public Object invoke(Object proxy, final Method method, Object[] args) {
         }
     }
 
+    private Flux<ByteBuffer> validateLength(final HttpRequest request) {
+        final Flux<ByteBuffer> bbFlux = request.body();
+        if (bbFlux == null) {
+            return Flux.empty();
+        }
+
+        return Flux.defer(() -> {
+            Long expectedLength = Long.valueOf(request.headers().value("Content-Length"));
+            final AtomicLong currentTotalLength = new AtomicLong(0L);
+            return bbFlux.doOnEach(s -> {
+                if (s.isOnNext()) {
+                    ByteBuffer byteBuffer = s.get();
+                    int currentLength = (byteBuffer == null) ? 0 : byteBuffer.remaining();
+                    currentTotalLength.getAndAdd(currentLength);
```

Comment:
AtomicLong introduces synchronization internally which is not needed when we're in onNext/onComplete. 

The guarantee that a reactive publisher provide is - it will always serialize the call to multiple onNext and final onComplete, so no concurrent execution of onNext means AtomicLong internal synchronization is unnecessary.  

If the reason for using AtomicLong is to enable mutation within the lambda, then we could use a long array with one element 

```
final long[] currentTotalLengh = new long[1];
```

then use `currentTotalLength[0]` wherever needed like:

```
currentTotalLength[0] = currentTotalLength[0] + currentLength;
```

Reply:
The AtomicLong locked on multi-thread. Is there any chance one reactor request on client side will invoke different thread?

Reply:
I will take the array one as you suggested.

Reply:
The onNext can be called from multiple threads depending on how upstream is designed to produce and emit values (e.g. each value computed in separate threads and emitted from same thread). But these onNext  calls never occur concurrently i.e. two invocation on onNext never overlaps, that's the guarantee that reactive manifesto promises.